Engine/PSScriptAnalyzer.psm1

```diff
@@ -20,9 +20,13 @@ if (($PSVersionTable.Keys -contains "PSEdition") -and ($PSVersionTable.PSEdition
 else {
     if ($PSVersionTable.PSVersion.Major -eq 3) {
         $binaryModuleRoot = Join-Path -Path $PSModuleRoot -ChildPath 'PSv3'
+        # Without this, PSSA tries to load this from $PSHome
+        Add-Type -Path "$binaryModuleRoot/Newtonsoft.Json.dll"
```

Comment:
This is where the Newtonsoft.Json assembly load was failing. From fuslogvw:

```
*** Assembly Binder Log Entry  (2/14/2019 @ 11:54:57 PM) ***

The operation failed.
Bind result: hr = 0x80070002. The system cannot find the file specified.

Assembly manager loaded from:  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\clr.dll
Running under executable  C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe
--- A detailed error log follows. 

=== Pre-bind state information ===
LOG: DisplayName = Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
 (Fully-specified)
LOG: Appbase = file:///C:/Windows/System32/WindowsPowerShell/v1.0/
LOG: Initial PrivatePath = NULL
LOG: Dynamic Base = NULL
LOG: Cache Base = NULL
LOG: AppName = powershell.exe
Calling assembly : (Unknown).
===
LOG: This bind starts in default load context.
LOG: No application configuration file found.
LOG: Using host configuration file: 
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config.
LOG: Post-policy reference: Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
LOG: GAC Lookup was unsuccessful.
LOG: Attempting download of new URL file:///C:/Windows/System32/WindowsPowerShell/v1.0/Newtonsoft.Json.DLL.
LOG: Attempting download of new URL file:///C:/Windows/System32/WindowsPowerShell/v1.0/Newtonsoft.Json/Newtonsoft.Json.DLL.
LOG: Attempting download of new URL file:///C:/Windows/System32/WindowsPowerShell/v1.0/Newtonsoft.Json.EXE.
LOG: Attempting download of new URL file:///C:/Windows/System32/WindowsPowerShell/v1.0/Newtonsoft.Json/Newtonsoft.Json.EXE.
LOG: All probing URLs attempted and failed.
```

What I think is happening is:
- We ship Newtonsoft.Json.dll and that worked fine when it was a direct dependency of something in the Rules assembly.
- When UseCompatibleCmdlets got it to load directly, it worked fine.
- **BUT**, UseCompatibleCommands now comes first, and it doesn't have a *direct* dependency on Newtonsoft.Json.dll; it depends on CrossCompatibility.dll, which in turn depends on Newtonsoft.Json.dll.
- Either MEF or PowerShell 3/4 assumes that 2nd-degree dependencies will be in the same directory as the main module/running executable and doesn't bother to check the dir where CrossCompatibility.dll is.
- My suspicion is with the older MEF; PowerShell is usually pretty good about finding and loading things.

Although my `Add-Type` solution works here, it's brittle insofar as other people who take external dependencies will hit the same problem. And it also subverts the theoretical model of lazy loading PSSA has (but to be honest, to support that model, I think work needs to be done to move away from MEF).